### ISSUE TITLE
no Gnome keyring access with python3-keyring v8.0 (and above)

### ISSUE THREAD

**Issue by akki42** [bit-team/backintime#534]
Hi there,

since my openSuse xfce system updated python3-keyring from version 7.3 to 8.0 (and above, current version in repo: 8.1.1), Gnome keyring access is broken (ssh keys and EncFS passwords are not stored or retrieved; 'save passwords in keyring' is greyed out).

According to [python3-keyring's changelog](https://pythonhosted.org/keyring/history.html) and the discussion at https://github.com/jaraco/keyring/issues/196, there was some major, backward-incompatible change in v8.0.

BiT's debug output shows

- with python3-keyring v.7.3:
`DEBUG: [common/tools.py:611 keyring_supported] Found appropriate keyring 'keyring.backends.Gnome`

- with python3-keyring v.8.1.1:
`DEBUG: [common/tools.py:611 keyring_supported] Found appropriate keyring 'keyring.backends.fail'`

- with python3-keyring v8.2 and keyrings.alt v .1.1 (installed via pip):
`DEBUG: [common/tools.py:611 keyring_supported] Found appropriate keyring 'keyrings.alt.Gnome'`

Unfortunately, in the latter case, though Gnome keyring seems to be available, the 'save passwords' checkbox remains greyed out.

Does that look like a BiT error to you - or rather some fault in my keyring or keyrings.alt installation?

Many thanks & best regards!

**Comment by Germar** [comment]
Those API changes make BIT incompatible at the moment. Good news is that those changes seems to make things easier. BIT don't need to check which keyring is used anymore because only those who should are available.

I'll make myself a TestVM with python3-keyring < v8.0 and fix it in the next days.

**Comment by Germar** [comment]
I set up a OpenSuse Tumbleweed VM and tested this. After installing `python3-secretstorage` I was able to access keyring even with `python3-keyring` v8.2 (from `devel:languages:python3`)

I didn't found `secretstorage` in repositories. So I installed it with `pip`

```
sudo zypper install autoconf gcc python3-devel
sudo pip3 install secretstorage
```

I used BackInTime 1.1.12 from source for testing. Not sure which version is in OpenSuse repositories...
